pandas/tests/indexes/test_base.py

```diff
@@ -1557,6 +1557,41 @@ def test_string_index_repr(self):
 
                 self.assertEqual(coerce(idx), expected)
 
+    def test_map(self):
+
+        # Applying a function to the Index
+        df = pd.DataFrame([[0, 1], [2, 3]], columns=['c1', 'c2'], index=['i1', 'i2'])
+        df.index.name = "Numbering"
+        df.index = df.index.map(lambda x: x.upper())
```

Comment:
you need to move the tests to ``pandas/tests/index/common.py/Base`` as I indicated above. Then you can follow the other test examples; it will be then run for each index type.